ragna/deploy/_ui/components/metadata_filters_builder.py

```diff
@@ -0,0 +1,275 @@
+import panel as pn
+import param
+
+from ragna.core._metadata_filter import MetadataFilter, MetadataOperator
+
+EXCLUDED_OPERATORS = ["RAW", "AND", "OR"]
+ALLOWED_OPERATORS = [
+    op
+    for op in list(MetadataOperator.__members__.keys())
+    if op not in EXCLUDED_OPERATORS
+]
+
+DUNDER_METHODS = {f"__{op}__".lower(): op for op in ALLOWED_OPERATORS}
```

Comment:
Can remove this validation if not helpful. It uses the type information passed by the source storage to infer a list of valid operators.

It's not super useful though as string will still allow 'GT' etc., which I don't think is helpful (e.g. `path > /my_path` is valid Python code, but I don't think it's what users will want).

See below (`compute_valid_operator_options` etc) for more details.